main.go

```diff
@@ -19,12 +19,22 @@ func main() {
 	var (
 		repo  string
 		image string
+		grepo string
 	)
 
 	ss := strings.Split(os.Args[1], "/")
 
 	if len(ss) > 2 {
-		repo = ss[0]
+		if strings.Contains(ss[0], "gcr.io") {
+			repo = "gcr.io"
+			grepo = ss[0]
+		} else {
+			repo = ss[0]
+		}
+		image = strings.Join(ss[1:], "/")
+	} else if strings.Contains(ss[0], "gcr.io") {
+		repo = "gcr.io"
+		grepo = ss[0]
```

Comment:
Should `grepo` be extracted in `gcr.go`? I think it's ok to replace existed switch cases:

```go
switch {
	case repo == "hub.docker.com":
		...
	case repo == "quay.io":
		...
	case strings.HasSuffix(repo, "gcr.io"):
		...
}
```